classes/submissionFile/DAO.inc.php

```diff
@@ -0,0 +1,464 @@
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+
+use PKP\core\EntityDAO;
+use PKP\db\DAORegistry;
+use PKP\plugins\PKPPubIdPluginDAO;
+use PKP\services\PKPSchemaService;
+use PKP\submission\reviewRound\ReviewRound;
+
+use stdClass;
+
+class DAO extends EntityDAO implements PKPPubIdPluginDAO
+{
+    /** @copydoc EntityDAO::$schema */
+    public $schema = PKPSchemaService::SCHEMA_SUBMISSION_FILE;
+
+    /** @copydoc EntityDAO::$table */
+    public $table = 'submission_files';
+
+    /** @copydoc EntityDAO::$settingsTable */
+    public $settingsTable = 'submission_file_settings';
+
+    /** @copydoc EntityDAO::$primaryKeyColumn */
+    public $primaryKeyColumn = 'submission_file_id';
+
+    /** @copydoc EntityDAO::$primaryTableColumns */
+    public $primaryTableColumns = [
+        'assocId' => 'assoc_id',
+        'assocType' => 'assoc_type',
+        'createdAt' => 'created_at',
+        'fileId' => 'file_id',
+        'fileStage' => 'file_stage',
+        'genreId' => 'genre_id',
+        'id' => 'submission_file_id',
+        'sourceSubmissionFileId' => 'source_submission_file_id',
+        'submissionId' => 'submission_id',
+        'updatedAt' => 'updated_at',
+        'uploaderUserId' => 'uploader_user_id',
+        'viewable' => 'viewable',
+    ];
+
+    /**
+     * Instantiate a new SubmissionFile
+     */
+    public function newDataObject(): SubmissionFile
+    {
+        return App::make(SubmissionFile::class);
+    }
+
+    /**
+     * @copydoc EntityDAO::get()
+     */
+    public function get(int $id): ?SubmissionFile
+    {
+        $row = DB::table($this->table . ' as sf')
+            ->leftJoin('submissions as s', 's.submission_id', '=', 'sf.submission_id')
+            ->leftJoin('files as f', 'f.file_id', '=', 'sf.file_id')
+            ->where($this->primaryKeyColumn, '=', $id)
+            ->select(['sf.*', 'f.*', 's.locale as locale'])
+            ->first();
+        return $row ? $this->fromRow($row) : null;
+    }
+
+    /**
+     * Get the number of announcements matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('a.' . $this->primaryKeyColumn)
+            ->pluck('a.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of announcements matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->leftJoin('submissions as s', 's.submission_id', '=', 'sf.submission_id')
+            ->leftJoin('files as f', 'f.file_id', '=', 'sf.file_id')
+            ->select(['sf.*', 'f.*', 's.locale as locale'])
```

Comment:
The `Collector` already has a `select()` clause specifying `sf.*` (or it should be once that is corrected -- see other comment), so I would suggest using `addSelect(['f.*', 's.locale as locale'])` here. But since these additional joins are required for `fromRow` to work, why not move them into the `getQueryBuilder` function entirely?